src/main/flight/mixer.c

```diff
@@ -346,14 +346,92 @@ static void applyFlipOverAfterCrashModeToMotors(void)
+            float acceleration = scaledRPMLimit - mixerRuntime.rpmLimiterPreviousRPMLimit;
+            if (acceleration > 0) {
+                acceleration = MIN(acceleration, mixerRuntime.rpmLimiterAccelerationLimit);
+                scaledRPMLimit = mixerRuntime.rpmLimiterPreviousRPMLimit + acceleration;
+            }
+        } 
+        else {
+            throttle = throttle * mixerRuntime.rpmLimiterExpectedThrottleLimit;
+            scaledRPMLimit = ((mixerConfig()->rpm_limiter_rpm_limit)) * 100.0f;
+        }
+        float smoothedRPMError = averageRPMSmoothed - scaledRPMLimit;
+        float rpm_limiterP = smoothedRPMError * mixerRuntime.rpmLimiterPGain; //+ when overspped
+        float rpmLimiterD = (smoothedRPMError - mixerRuntime.rpmLimiterPreviousSmoothedRPMError) * mixerRuntime.rpmLimiterDGain; // + when quickly going overspeed
+        if (mixerConfig()->rpm_limiter_rpm_linearization) {
+            //don't let I term wind up if throttle is below the motor idle
+            if (rcCommandThrottle < motorConfig()->digitalIdleOffsetValue / 10000.0f) {
+                mixerRuntime.rpmLimiterI *= 1.0f / (1.0f + (pidGetDT() * 10.0f)); //slowly ramp down i term instead of resetting to avoid throttle pulsing cheats
+            } else if (!motorsSaturated) {
+                //don't let I term wind up if motors are saturated. Otherwise, motors may stay at high throttle even after low throttle is commanded
+                mixerRuntime.rpmLimiterI += smoothedRPMError * mixerRuntime.rpmLimiterIGain;
+            }
+            pidOutput = rpm_limiterP + mixerRuntime.rpmLimiterI + rpmLimiterD;
+        } else {
+            throttle = throttle * mixerRuntime.rpmLimiterExpectedThrottleLimit;
+            mixerRuntime.rpmLimiterI += smoothedRPMError * mixerRuntime.rpmLimiterIGain;
+            mixerRuntime.rpmLimiterI = MAX(mixerRuntime.rpmLimiterI, 0.0f);
+            pidOutput = rpm_limiterP + mixerRuntime.rpmLimiterI + rpmLimiterD;
+            if (pidOutput > 0.05) {
+                mixerRuntime.rpmLimiterExpectedThrottleLimit = 0.9994 * mixerRuntime.rpmLimiterExpectedThrottleLimit;
+            }
+            if (pidOutput < -0.05 && rcCommand[THROTTLE] > 1950 && !motorsSaturated) {
+                mixerRuntime.rpmLimiterExpectedThrottleLimit = 1.0006f * mixerRuntime.rpmLimiterExpectedThrottleLimit;
+                mixerRuntime.rpmLimiterExpectedThrottleLimit = MAX(mixerRuntime.rpmLimiterExpectedThrottleLimit, 1.0f);
+            }
+            pidOutput = MAX(pidOutput, 0.0f);
+        }
+        if (mixerRuntime.rpmLimiterInit) {
+            if (mixerConfig()->rpm_limiter_rpm_linearization) {
+                throttle = constrainf(-pidOutput, 0.01f, 1.0f);
+            } else {
+                throttle = constrainf(throttle-pidOutput, 0.01f, 1.0f);
```

Comment:
really, throttle should be allowed to go all the way down to 0.0

Reply:
This was supposed to fix an issue headsup was having with two of his motors stopping completely when arming and rolling while still on the launch blocks. I can't reproduce it now though so I'll assume his dshot_idle_rpm is set wrong. 
This is something that dynamic idle does on line 648 as well
`throttle = MAX(throttle, 0.01f);`